File: Cryptographic_Frontend/src/methods/GenerateKeyPairMethod.cpp
#include <methods/GenerateKeyPairMethod.hpp>

#include <StringArgument.hpp>
#include <IntegerArgument.hpp>
#include <ResponseMessage.hpp>
#include <json/json.h>

using namespace cf;
using std::string;

GenerateKeyPairMethod::GenerateKeyPairMethod(
        string keyType,
        int keySize,
        string publicExponent)
    : Method("GenerateKeyPair")
{
    /* methodMessage es dueño de los argumentos */
    addArgument(ArgumentPtr(new StringArgument("keyType", keyType)));
    addArgument(ArgumentPtr(new IntegerArgument("keySize", keySize)));
    addArgument(ArgumentPtr(new StringArgument("publicExponent", publicExponent)));
}

ResponseMessageStrategy GenerateKeyPairMethod::getStrategy() const
{
    return [](std::string const & message) -> ResponseMessagePtr {
        Json::Value json;
        Json::Reader reader;

        reader.parse(message, json);
        ResponseMessagePtr response(new ResponseMessage());

        response->setValue("handler", json["handler"].asInt());

        return std::move(response);
    };
}
// kate: indent-mode cstyle; indent-width 4; replace-tabs on; 

File: Cryptographic_Backend/src/main/java/cl/niclabs/cb/backend/methods/GenerateKeyPairMethod.java
package cl.niclabs.cb.backend.methods;

import cl.niclabs.cb.backend.Method;

public abstract class GenerateKeyPairMethod implements Method {

  public static class ReturnValue {
    public final String handler;

      public ReturnValue(String handler) {
          this.handler = handler;
      }
  }

  public static class Args {
    public final String keyType;
    public final int keySize;
    public final String publicExponent;

      public Args(String keyType, int keySize, String publicExponent) {
          this.keyType = keyType;
          this.keySize = keySize;
          this.publicExponent = publicExponent;
      }
  }

  protected final String keyType;
  protected final int keySize;
  protected final String publicExponent;

    public GenerateKeyPairMethod(Args args) {
    this.keyType = args.keyType;
    this.keySize = args.keySize;
    this.publicExponent = args.publicExponent;
  }

}
